src/wallet/wallet.cpp

```diff
@@ -3673,6 +3668,33 @@ bool CMerkleTx::AcceptToMemoryPool(bool fLimitFree, bool fRejectAbsurdFee)
     return ::AcceptToMemoryPool(mempool, state, *this, fLimitFree, NULL, fRejectAbsurdFee);
 }
 
+void CWallet::GetFilteredTransactions(std::multimap<int64_t, CWalletTx >& outEntries, const std::string& address)
+{
+    LOCK2(cs_main, cs_wallet);
+    CScript scriptPubKey;
+
+    if(address.compare("*")!=0){
+        CBitcoinAddress baddress = CBitcoinAddress(address);
+        scriptPubKey = GetScriptForDestination(baddress.Get(), false);
+    }
+
+    //getting all Txes of address in the wallet
+    for (auto & p : mapWallet) {
+           CWalletTx wtx = p.second;
+           if(address.compare("*")==0) {
```

Comment:
It would be more efficient to use a boolean, since this comparison has taken place above.